app/shop.py

```diff
@@ -0,0 +1,50 @@
+import datetime
+from typing import Any
+
+
+class Shop:
+    def __init__(self,
+                 name: str,
+                 location: list,
+                 products: dict) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+
+    def sum_of_products(self, product: dict) -> float:
+        sum_of_products = (
+            self.products["milk"] * product["milk"]
+            + self.products["bread"] * product["bread"]
+            + self.products["butter"] * product["butter"]
+        )
+        return sum_of_products
+
+    def customers_trip(self,
+                       customer: str,
+                       func: Any,
+                       function: Any) -> None:
+        print(f"{customer}'s trip to the {self.name} costs {func + function}")
+
+    def buy_products(self,
+                     product: dict,
+                     func: Any,
+                     customer: str) -> None:
+        dt = datetime.datetime.now()
+        dt = dt.replace(year=2021,
```

Comment:
Why do you use it? But if you need, you can use `datetime.datetime(year=..., month=...)`